feat(client): add WebSocket channel setup

lorenzo-cavazzi opened the pull request:
This PR adds the client-side logic for WebSocket.

The WebSocket channel is set up and it gets re-created in case of recoverable errors (to be further expanded), and it listens to incoming messages.
Once a new message comes, it checks for a suitable handler and executes it.
Mind that we save minimal data to the store for the WebSocket channel itself, but the handlers can modify any part of the model.

This fully moves the logic to check for UI version updates to the WebSocket channel. It's also a good example of how we can send all the required data through the channel, although we probably want to avoid this behavior in most cases where we need to get a lot of data.

**How to test**
Open the network tab and verify that the WebSocket channel is created and information about the client version is sent regularly.

![Screenshot_20220829_173034](https://user-images.githubusercontent.com/43481553/187237957-a26aa1aa-5217-421c-a019-e3064535bfa2.png)

**Further development**
There will be a follow-up issue to finish and enable the (already partially coded) logic for finding session updates through the WebSocket channel.

/deploy renku-gateway=master #persist
fix #1338

ciyer reviewed (changes requested):
This looks very good!

I have a few minor comments on some code improvements, but I overall like the structure.

lorenzo-cavazzi commented on client/src/websocket/handlers/userHandlers.ts:
```diff
@@ -0,0 +1,57 @@
+function handleUserInit(data: Record<string, unknown>, webSocket: WebSocket, model: any): boolean {
+  if (data?.message && (data.message as string).toLowerCase().includes("connection established")) {
+    // send request for getting the UI version
+    webSocket.send(JSON.stringify(new WsMessage({ requestServerVersion: true }, "init")));
```
This pattern is repeated in every `send` function of both the client and server WebSocket channel. I changed it everywhere so that all the `JSON.stringify(<content>)` become `(<content>).toString()`.

lorenzo-cavazzi commented:
Thank you for the review! I addresses the points you mentioned, and improved the logic to open/close the channel so that we won't miss the session cookies (or retry shortly after when that happens)

ciyer reviewed (dismissed):
This is great and works well. The occasional initialization problem on first load seems to be fixed now. Looking forward to using this to improve the UX!

lorenzo-cavazzi commented:
Thanks for the review!
I ended up force-pushing to this branch to keep 2 separate commits merged to master (WebSocket infrastructure vs moving the UI version check to WebSocket) instead of squashing the 10+ commits together.

Could you re-approve?
